File: resources/android/src/com/victorycodedev/plugins/toastkit/model/ToastKitModels.kt
package com.victorycodedev.plugins.toastkit.model

import androidx.compose.ui.graphics.Color

internal enum class ToastKitPosition { TOP, CENTER, BOTTOM }
internal enum class ToastKitAnimation { FADE, SLIDE, SCALE, SPRING, SNAP, POP, REVEAL, BOUNCE }
internal enum class ToastKitDirection { AUTO, LEFT, RIGHT, TOP, BOTTOM }
internal enum class ToastKitStrategy { QUEUE, STACK }
internal enum class ToastKitTextSize { XS, SM, BASE, LG, XL }
internal enum class ToastKitTextWeight { NORMAL, MEDIUM, SEMIBOLD, BOLD }
internal enum class ToastKitTextAlign { LEFT, CENTER, RIGHT }

internal data class ToastKitIcon(val name: String?, val android: String?)
internal data class ToastKitAction(val id: String, val label: String)
internal data class ToastKitTypography(
    val font: String? = null,
    val size: ToastKitTextSize? = null,
    val weight: ToastKitTextWeight? = null,
    val align: ToastKitTextAlign? = null,
    val italic: Boolean? = null,
)
internal data class ToastKitStyle(
    val background: Color,
    val foreground: Color,
    val iconColor: Color,
    val actionColor: Color,
    val cornerRadius: Float,
    val padding: Float,
    val shadow: Boolean,
)

internal data class ToastKitConfiguration(
    val id: String,
    val uniqueKey: String?,
    val message: String,
    val title: String?,
    val variant: String,
    val icon: ToastKitIcon?,
    val position: ToastKitPosition,
    val durationMs: Long?,
    val animation: ToastKitAnimation,
    val direction: ToastKitDirection,
    val progress: Float?,
    val loading: Boolean,
    val swipeToDismiss: Boolean,
    val dismissible: Boolean,
    val action: ToastKitAction?,
    val style: ToastKitStyle,
    val strategy: ToastKitStrategy,
    val maxVisible: Int,
    val text: ToastKitTypography? = null,
    val titleText: ToastKitTypography? = null,
)

internal data class ToastKitState(
    val configuration: ToastKitConfiguration,
    val visible: Boolean = false,
    val terminated: Boolean = false,
)

File: resources/ios/Sources/Model/ToastKitModels.swift
import SwiftUI

enum ToastKitPosition: String { case top, center, bottom }
enum ToastKitAnimationKind: String { case fade, slide, scale, spring, snap, pop, reveal, bounce }
enum ToastKitDirection: String { case auto, left, right, top, bottom }
enum ToastKitStrategy: String { case queue, stack }
enum ToastKitTextSize: String { case xs, sm, base, lg, xl }
enum ToastKitTextWeight: String { case normal, medium, semibold, bold }
enum ToastKitTextAlign: String { case left, center, right }

struct ToastKitIconConfiguration {
    let name: String?
    let ios: String?
}

struct ToastKitActionConfiguration {
    let id: String
    let label: String
}

struct ToastKitTypographyConfiguration {
    var font: String?
    var size: ToastKitTextSize?
    var weight: ToastKitTextWeight?
    var align: ToastKitTextAlign?
    var italic: Bool?
}

struct ToastKitStyleConfiguration {
    var background: Color
    var foreground: Color
    var iconColor: Color
    var actionColor: Color
    var cornerRadius: CGFloat
    var padding: CGFloat
    var shadow: Bool
}

struct ToastKitConfiguration: Identifiable {
    let id: String
    let uniqueKey: String?
    var nativeIos: Bool
    var nativeAndroid: Bool
    var message: String
    var title: String?
    var variant: String
    var icon: ToastKitIconConfiguration?
    var position: ToastKitPosition
    var durationMilliseconds: Int?
    var animation: ToastKitAnimationKind
    var direction: ToastKitDirection
    var progress: Double?
    var loading: Bool
    var swipeToDismiss: Bool
    var dismissible: Bool
    var action: ToastKitActionConfiguration?
    var style: ToastKitStyleConfiguration
    var strategy: ToastKitStrategy
    var maxVisible: Int
    var text: ToastKitTypographyConfiguration?
    var titleText: ToastKitTypographyConfiguration?
}

struct ToastKitState {
    var configuration: ToastKitConfiguration
    var visible = false
    var terminated = false
}
